### DISCUSSION TITLE
Delta E (2000) between two images?

### DISCUSSION THREAD

**Discussion by blue-j** [colour-science/colour#1011] [Q&A] [answered]
I'm feeding two images into the color difference function, and then calculating stats:

`delta_E = colour.delta_E(image1_lab, image2_lab, method='CIE 2000')

print(np.min(delta_E),np.max(delta_E),np.mean(delta_E),np.std(delta_E))`

It seems to work fine.  But is this a valid use of the function?  It seems built to send two Lab value not two Lab images.

Thanks!
J

**Comment by thomasmansencal** [comment, marked as answer]
Yes this is fine, the function is vectorised so you can certainly pass an image!

**Reply by blue-j** [reply]
Thanks so much!  What a relief. Is there any way to get the raw delta E values that are being calculated?

**Reply by thomasmansencal** [reply]
Not sure what you mean by the *raw delta E values*?

**Reply by blue-j** [reply]
I perhaps have made an incorrect assumption. I assumed you'd have to calculate a bunch of separate deltas? I'm asking for and receiving a mean, for example, so it's got to be a mean OF something, yes?  Please forgive me if I'm being daft, I'm a lifelong student and grateful to those who share their knowledge. 🙏

**Reply by thomasmansencal** [reply]
No worries!

Say you have two images `a` an `b` in _CIE Lab_ with the following shape: (600, 800, 3). When you compute the Delta E between the two, the resulting array should be of the (600, 800) shape and it represent per-pixel Delta E.

Does it make sense?

**Reply by blue-j** [reply]
It does! Thank you so much for your help and for your sharing this incredible work.
